src/main/java/com/exasol/adapter/dynamodb/mapping/ToDecimalPropertyToColumnValueExtractor.java

```diff
@@ -0,0 +1,83 @@
+import com.exasol.sql.expression.BigDecimalLiteral;
+import com.exasol.sql.expression.NullLiteral;
+import com.exasol.sql.expression.ValueExpression;
+
+/**
+ * This class extracts DECIMAL values from document data like. The extraction is defined using a
+ * {@link ToDecimalPropertyToColumnMapping}.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public abstract class ToDecimalPropertyToColumnValueExtractor<DocumentVisitorType>
+        extends AbstractPropertyToColumnValueExtractor<DocumentVisitorType> {
+    private final ToDecimalPropertyToColumnMapping column;
+
+    /**
+     * Create an instance of {@link ToDecimalPropertyToColumnValueExtractor}
+     *
+     * @param column {@link ToDecimalPropertyToColumnMapping} defining the mapping
+     */
+    public ToDecimalPropertyToColumnValueExtractor(final ToDecimalPropertyToColumnMapping column) {
+        super(column);
+        this.column = column;
+    }
+
+    @Override
+    protected final ValueExpression mapValue(final DocumentNode<DocumentVisitorType> documentValue) {
+        final BigDecimal decimalValue = mapValueToDecimalWithExceptionHandling(documentValue);
+        if (decimalValue == null) {
+            return handleNotANumber();
+        } else {
+            final BigDecimal decimalWithDestinationScale = decimalValue.setScale(this.column.getDecimalScale(),
+                    RoundingMode.FLOOR);
+            if (decimalWithDestinationScale.precision() > this.column.getDecimalPrecision()) {
+                return handleOverflow();
+            } else {
+                return BigDecimalLiteral.of(decimalWithDestinationScale);
+            }
+        }
+    }
+
+    private BigDecimal mapValueToDecimalWithExceptionHandling(final DocumentNode<DocumentVisitorType> documentValue) {
+        try {
+            return mapValueToDecimal(documentValue);
+        } catch (final NumberFormatException exception) {
+            throw new ColumnValueExtractorException(
+                    "Could not read map the data source's number. Cause: " + exception.getMessage(), exception,
+                    this.column);
+        }
+    }
+
+    private ValueExpression handleNotANumber() {
+        if (this.column.getNotANumberBehaviour() == MappingErrorBehaviour.ABORT) {
+            throw new ColumnValueExtractorException(
+                    "The input value was no number. Try using a different mapping or ignore this error by setting notANumberBehaviour = \"null\".",
+                    this.column);
+        } else {
+            return NullLiteral.nullLiteral();
+        }
+    }
+
+    private ValueExpression handleOverflow() {
+        if (this.column.getOverflowBehaviour() == MappingErrorBehaviour.ABORT) {
+            throw new OverflowException(
+                    "Decimal overflow. You can either increase the DECIMAL precision of this column or set the overflow behaviour to NULL.",
```

Comment:
This comment about setting something somehow different from the previous one. Please, unify them

Reply:
I tied